lib/websocket/websocket-handler.ts

```diff
@@ -0,0 +1,236 @@
+            }
+
+            var sub = new Subscription(s.correlationId,
+                                            s.security,
+                                            s.fields,
+                                            s.options);
+            subscriptions.push(sub);
+            receivedSubscriptions.add(sub);
+
+            // Add event listener for each subscription
+            sub.on('data', (data : any) : void => {
+// For debug purpose
+var ts = process.hrtime();
+socket.log.debug({data: {cid: sub.correlationId, time: ts}}, 'Data received');
+data['DEBUG_TIME'] = ts[0] + '.' + ts[1];
+
+                // Emit the data
+                // TODO: Acknowledgement callback function?
+                socket.emit('data', {'correlationId' : sub.correlationId,
+                                     'data' : data});
+                socket.log.info('Data sent');
+            });
+        });
+
+        // Subscribe user request through blpapi-wrapper
+        socket.blpSession.subscribe(subscriptions)
+        .then((): void => {
+            if (socket.isConnected()) {
+                subscriptions.forEach((s: Subscription): void => {
+                    activeSubscriptions.add(s);
+                });
+                socket.log.debug('Subscribed');
+                socket.emit('subscribed');
+            } else { // Unsubscribe if socket already closed
+                try {
+                    socket.blpSession.unsubscribe(subscriptions);
+                } catch (ex) {
+                    socket.log.error(ex, 'Error Unsubscribing');
+                }
+                subscriptions.forEach((s: Subscription): void => {
+                    s.removeAllListeners();
+                    receivedSubscriptions.delete(s);
+                });
+                socket.log.debug('Unsubscribed all active subscriptions');
+            }
+        })
+        .catch( (err: Error): void => {
+            socket.log.error(err, 'Error Subscribing');
+            subscriptions.forEach((s: Subscription): void => {
+                s.removeAllListeners();
+                receivedSubscriptions.delete(s);
+            });
+            if (socket.isConnected()) {
+                socket.emit('err', err);
+            }
+        });
+    });
+
+    // Unsubscribe
+    socket.on('unsubscribe', (data : { 'correlationIds': number[] }): void => {
+        socket.log.info('Unsubscribe request received');
+
+        if (!activeSubscriptions.size) {
+            socket.log.debug('No active subscriptions.');
+            socket.emit('err', { message: 'No active subscriptions.' });
+            return;
+        }
+
+        var subscriptions : Subscription[] = [];
+
+        // If no correlation Id specified
+        // the default behavior is to unsubscribe all active subscriptions
+        if (!data
+            || !_.has(data, 'correlationIds')
+            || !_.isArray(data.correlationIds)
+            || !data.correlationIds.length) {
```

Comment:
I move the logical operation to the end of line. However, tslint doesn't seem to like `{` onto the next line?